core/server/api_container/server/startosis_engine/kurtosis_types/service_config/service_config.go

```diff
@@ -268,6 +275,22 @@ func (config *ServiceConfig) ToKurtosisType() (*kurtosis_core_rpc_api_bindings.S
 	return builder.Build(), nil
 }
 
+func (config *ServiceConfig) GetReadyConditions() (*ReadyConditions, *startosis_errors.InterpretationError) {
```

Comment:
Where is this called?

By doing that, you're breaking the framework idea of "fail as early as possible". By having this `ValidateAndFillReadyConditions` in the ReadyConditions object, you're basically postponing parsing of the argument to a later code execution. For example if we pass in an invalid duration string, it currently doesn't fail at interpretation, which is sad because it could very well do.
Therefore in here you have to create this `GetReadyConditions` method, which IDK where it's being used ATM but I basically you're propagating the same behaviour: interpretation of ServiceConfig will not fail even if ReadyConditions is invalid.

I think the expected way to implement this is:
- Do not store any of the timeout, intrval, etc inside ReadyConditions
- Add validator to each field, such that interpretation will validate the fields
- Do the cast inside each Getter you have in ReadyConditions.

If you do all this, I don't think it will be more code, and potential user input failures will happen earlier.
Happy to chat if you have concerns, maybe I'm missing something